lib/venue.rb

```diff
@@ -1,2 +1,34 @@
 class Venue
+  attr_reader :name,
+              :capacity,
+              :patrons
+  def initialize(name, capacity)
+    @name     = name
+    @capacity = capacity
+    @patrons = []
+  end
+
+  def add_patron(patron)
+    @patrons << patron
+  end
+
+  def yell_at_patrons
+    @patrons.map do |patron|
+      patron.upcase
+    end
+  end
+
+  def over_capacity?
+    if @patrons.count > capacity
+      true
+    else
+      false
+    end
+  end
+
+  def kick_out
+    if self.over_capacity? == true
```

Comment:
Great job Chris! In fact, this actually works without having to run the `if` statement to test the `over_capacity?`; you can call `@patrons.shift until self.over_capacity? == false` on its own. Comment out lines 30 and 32 and see for yourself!